File: generators/dwarfWowNames.py
__all__ = ['dwarfWowNames']

# Don't look below, you will not understand this Python code :) I don't.

from js2py.pyjs import *
# setting scope
var = Scope( JS_BUILTINS )
set_global_object(var)

# Code follows:
var.registers(['nm1', 'nm4', 'nameGen', 'nm5', 'nm8', 'nm3', 'nm7', 'nm2', 'nm6'])
@Js
def PyJsHoisted_nameGen_(type, this, arguments, var=var):
    var = Scope({'arguments':arguments, 'this':this, 'type':type}, var)
    var.registers(['tp', 'result', 'type'])
    var.put('tp', var.get('type'))
    var.put('result', Js([]))
    #for JS loop
    var.put('i', Js(0.0))
    while (var.get('i')<Js(10.0)):
        try:
            var.put('rnd4', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm7').get('length'))))
            var.put('rnd5', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm8').get('length'))))
            if PyJsStrictEq(var.get('tp'),Js(1.0)):
                var.put('rnd', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm4').get('length'))))
                var.put('rnd2', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm5').get('length'))))
                var.put('rnd3', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm6').get('length'))))
                var.put('names', (((((var.get('nm4').get(var.get('rnd'))+var.get('nm5').get(var.get('rnd2')))+var.get('nm6').get(var.get('rnd3')))+Js(' '))+var.get('nm7').get(var.get('rnd4')))+var.get('nm8').get(var.get('rnd5'))))
            else:
                var.put('rnd', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm1').get('length'))))
                var.put('rnd2', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm2').get('length'))))
                var.put('rnd3', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm3').get('length'))))
                var.put('names', (((((var.get('nm1').get(var.get('rnd'))+var.get('nm2').get(var.get('rnd2')))+var.get('nm3').get(var.get('rnd3')))+Js(' '))+var.get('nm7').get(var.get('rnd4')))+var.get('nm8').get(var.get('rnd5'))))
            var.get('result').callprop('push', var.get('names'))
        finally:
                (var.put('i',Js(var.get('i').to_number())+Js(1))-Js(1))
    pass
    return var.get('result')
PyJsHoisted_nameGen_.func_name = 'nameGen'
var.put('nameGen', PyJsHoisted_nameGen_)
var.put('nm1', Js([Js('Am'), Js('Arm'), Js('Bal'), Js('Ban'), Js('Bar'), Js('Bel'), Js('Beng'), Js('Bhal'), Js('Bhar'), Js('Bhel'), Js('Bram'), Js('Bran'), Js('Brom'), Js('Brum'), Js('Bun'), Js('Dal'), Js('Dar'), Js('Dol'), Js('Dram'), Js('Drom'), Js('Drum'), Js('Dul'), Js('Em'), Js('Erm'), Js('Gal'), Js('Gar'), Js('Ger'), Js('Gim'), Js('Gir'), Js('Gol'), Js('Gor'), Js('Gral'), Js('Gram'), Js('Gran'), Js('Grem'), Js('Gren'), Js('Gril'), Js('Grim'), Js('Grom'), Js('Grul'), Js('Grum'), Js('Grun'), Js('Gry'), Js('Gul'), Js('Har'), Js('Hjal'), Js('Hjol'), Js('Hjul'), Js('Hor'), Js('Hulf'), Js('Hur'), Js('Irom'), Js('Iron'), Js('Jar'), Js('Jor'), Js('Kar'), Js('Khar'), Js('Kram'), Js('Krom'), Js('Krum'), Js('Mag'), Js('Mal'), Js('Mel'), Js('Muir'), Js('Mur'), Js('Ol'), Js('Orim'), Js('Orm'), Js('Rag'), Js('Reg'), Js('Rot'), Js('Shel'), Js('Sog'), Js('Sogn'), Js('Sug'), Js('Sugn'), Js('Thal'), Js('Thar'), Js('Thel'), Js('Ther'), Js('Tho'), Js('Thor'), Js('Thul'), Js('Thur'), Js('Thy'), Js('Tor'), Js('Ty'), Js('Um'), Js('Urm')]))
var.put('nm2', Js([Js('o'), Js('a'), Js('u'), Js('i'), Js('ia'), Js('iu'), Js('ou'), Js('ua'), Js('ah'), Js('uh'), Js('oh'), Js('ihr'), Js('ahr'), Js('ohr'), Js(''), Js(''), Js(''), Js(''), Js(''), Js(''), Js(''), Js(''), Js(''), Js(''), Js(''), Js(''), Js(''), Js(''), Js(''), Js(''), Js(''), Js(''), Js(''), Js(''), Js(''), Js(''), Js(''), Js('')]))
var.put('nm3', Js([Js('brek'), Js('dahr'), Js('dan'), Js('dar'), Js('dir'), Js('dohr'), Js('donore'), Js('dor'), Js('dram'), Js('dren'), Js('drom'), Js('drum'), Js('drus'), Js('duhr'), Js('dur'), Js('dus'), Js('edon'), Js('g'), Js('garn'), Js('gas'), Js('gorn'), Js('gos'), Js('gram'), Js('gran'), Js('grom'), Js('gron'), Js('grum'), Js('grun'), Js('gurn'), Js('gus'), Js('han'), Js('hann'), Js('hun'), Js('hunn'), Js('iggs'), Js('kahm'), Js('kam'), Js('kohm'), Js('kom'), Js('kuhm'), Js('kum'), Js('kyl'), Js('m'), Js('man'), Js('mand'), Js('mar'), Js('mek'), Js('miir'), Js('min'), Js('mir'), Js('mond'), Js('mor'), Js('mun'), Js('mund'), Js('mur'), Js('mus'), Js('myl'), Js('myr'), Js('n'), Js('nam'), Js('nar'), Js('ni'), Js('nik'), Js('nir'), Js('nom'), Js('nu'), Js('num'), Js('nur'), Js('nus'), Js('nyl'), Js('ram'), Js('ren'), Js('rig'), Js('rigg'), Js('rim'), Js('rom'), Js('ron'), Js('rum'), Js('rus'), Js('ruuk'), Js('ryl'), Js('tharm'), Js('tharn'), Js('thorm'), Js('thorn'), Js('thran'), Js('throm'), Js('thron'), Js('thrum'), Js('thrun'), Js('thurm'), Js('thurn'), Js('thus'), Js('yth')]))
var.put('nm4', Js([Js('Ang'), Js('Angr'), Js('Bel'), Js('Bell'), Js('Bon'), Js('Bonn'), Js('Braen'), Js('Bral'), Js('Brall'), Js('Bran'), Js('Bril'), Js('Brill'), Js('Bren'), Js('Bret'), Js('Brett'), Js('Brol'), Js('Broll'), Js('Bron'), Js('Brul'), Js('Brull'), Js('Bryl'), Js('Bryll'), Js('Bryn'), Js('Bryt'), Js('Byl'), Js('Byll'), Js('Daer'), Js('Dear'), Js('Dim'), Js('Ed'), Js('Ein'), Js('Eyn'), Js('Gwan'), Js('Gwen'), Js('Gwin'), Js('Gwyn'), Js('Gim'), Js('Gym'), Js('Ing'), Js('Ingr'), Js('Jen'), Js('Jenn'), Js('Jin'), Js('Jinn'), Js('Jyn'), Js('Jynn'), Js('Kait'), Js('Kar'), Js('Kat'), Js('Ket'), Js('Las'), Js('Lass'), Js('Les'), Js('Less'), Js('Lyes'), Js('Lys'), Js('Lyss'), Js('Maev'), Js('Mar'), Js('Mis'), Js('Mist'), Js('Myr'), Js('Mys'), Js('Myss'), Js('Myst'), Js('Nal'), Js('Nas'), Js('Nass'), Js('Nes'), Js('Ness'), Js('Nis'), Js('Niss'), Js('Nys'), Js('Nyss'), Js('Ong'), Js('Ongr'), Js('Ov'), Js('Raen'), Js('Ran'), Js('Red'), Js('Reyn'), Js('Run'), Js('Ryn'), Js('Sar'), Js('Sol'), Js('Tas'), Js('Taz'), Js('Tis'), Js('Tish'), Js('Tiz'), Js('Tys'), Js('Tysh'), Js('Ung'), Js('Ungr')]))
var.put('nm5', Js([Js('o'), Js('a'), Js('i'), Js('ia'), Js('io'), Js('ou'), Js('ua'), Js('ah'), Js('um'), Js('oh'), Js('ir'), Js('ar'), Js('or'), Js('un'), Js('eo'), Js('ea'), Js('in'), Js('im'), Js('an'), Js('am'), Js(''), Js(''), Js(''), Js(''), Js('')]))
var.put('nm6', Js([Js('leil'), Js('nar'), Js('nyl'), Js('myl'), Js('lyl'), Js('dyl'), Js('ris'), Js('ras'), Js('res'), Js('ros'), Js('win'), Js('wyn'), Js('waen'), Js('whin'), Js('whyn'), Js('whaen'), Js('tin'), Js('tyn'), Js('lyn'), Js('lin'), Js('lynn'), Js('linn'), Js('wynn'), Js('va'), Js('van'), Js('rin'), Js('ryn'), Js('ryl'), Js('nas'), Js('nan'), Js('ria'), Js('nia'), Js('ri'), Js('rip'), Js('nip'), Js('dora'), Js('leen'), Js('len'), Js('ma'), Js('la'), Js('mora'), Js('mura'), Js('mera'), Js('nura'), Js('nera'), Js('nora'), Js('glia'), Js('glian'), Js('giel'), Js('thiel'), Js('diel'), Js('thel'), Js('nis'), Js('niss'), Js('nys'), Js('nyss')]))
var.put('nm7', Js([Js('Barley'), Js('Battle'), Js('Black'), Js('Bone'), Js('Boulder'), Js('Bright'), Js('Bronze'), Js('Cask'), Js('Cliff'), Js('Cold'), Js('Crag'), Js('Dark'), Js('Deep'), Js('Dirge'), Js('Doom'), Js('Fire'), Js('Flint'), Js('Frost'), Js('Fuse'), Js('Gold'), Js('Hammer'), Js('High'), Js('Iron'), Js('Long'), Js('Marble'), Js('Molten'), Js('Mountain'), Js('Pale'), Js('Red'), Js('Slate'), Js('Snow'), Js('Steel'), Js('Stern'), Js('Stone'), Js('Storm'), Js('Stout'), Js('Thunder')]))
var.put('nm8', Js([Js('arm'), Js('axe'), Js('beard'), Js('belly'), Js('blade'), Js('braid'), Js('branch'), Js('brand'), Js('breaker'), Js('brew'), Js('brow'), Js('crag'), Js('dust'), Js('fall'), Js('fist'), Js('flayer'), Js('forge'), Js('fury'), Js('gem'), Js('grip'), Js('hammer'), Js('hand'), Js('helm'), Js('mane'), Js('mantle'), Js('ore'), Js('pike'), Js('river'), Js('roar'), Js('rock'), Js('shaper'), Js('shield'), Js('shout'), Js('steel'), Js('stone'), Js('toe')]))
pass
pass


# Add lib to the module scope
dwarfWowNames = var.to_python()

File: js/dwarfWowNames.js
var nm1 = ["Am","Arm","Bal","Ban","Bar","Bel","Beng","Bhal","Bhar","Bhel","Bram","Bran","Brom","Brum","Bun","Dal","Dar","Dol","Dram","Drom","Drum","Dul","Em","Erm","Gal","Gar","Ger","Gim","Gir","Gol","Gor","Gral","Gram","Gran","Grem","Gren","Gril","Grim","Grom","Grul","Grum","Grun","Gry","Gul","Har","Hjal","Hjol","Hjul","Hor","Hulf","Hur","Irom","Iron","Jar","Jor","Kar","Khar","Kram","Krom","Krum","Mag","Mal","Mel","Muir","Mur","Ol","Orim","Orm","Rag","Reg","Rot","Shel","Sog","Sogn","Sug","Sugn","Thal","Thar","Thel","Ther","Tho","Thor","Thul","Thur","Thy","Tor","Ty","Um","Urm"];
var nm2 = ["o","a","u","i","ia","iu","ou","ua","ah","uh","oh","ihr","ahr","ohr","","","","","","","","","","","","","","","","","","","","","","","",""];
var nm3 = ["brek","dahr","dan","dar","dir","dohr","donore","dor","dram","dren","drom","drum","drus","duhr","dur","dus","edon","g","garn","gas","gorn","gos","gram","gran","grom","gron","grum","grun","gurn","gus","han","hann","hun","hunn","iggs","kahm","kam","kohm","kom","kuhm","kum","kyl","m","man","mand","mar","mek","miir","min","mir","mond","mor","mun","mund","mur","mus","myl","myr","n","nam","nar","ni","nik","nir","nom","nu","num","nur","nus","nyl","ram","ren","rig","rigg","rim","rom","ron","rum","rus","ruuk","ryl","tharm","tharn","thorm","thorn","thran","throm","thron","thrum","thrun","thurm","thurn","thus","yth"];
var nm4 = ["Ang","Angr","Bel","Bell","Bon","Bonn","Braen","Bral","Brall","Bran","Bril","Brill","Bren","Bret","Brett","Brol","Broll","Bron","Brul","Brull","Bryl","Bryll","Bryn","Bryt","Byl","Byll","Daer","Dear","Dim","Ed","Ein","Eyn","Gwan","Gwen","Gwin","Gwyn","Gim","Gym","Ing","Ingr","Jen","Jenn","Jin","Jinn","Jyn","Jynn","Kait","Kar","Kat","Ket","Las","Lass","Les","Less","Lyes","Lys","Lyss","Maev","Mar","Mis","Mist","Myr","Mys","Myss","Myst","Nal","Nas","Nass","Nes","Ness","Nis","Niss","Nys","Nyss","Ong","Ongr","Ov","Raen","Ran","Red","Reyn","Run","Ryn","Sar","Sol","Tas","Taz","Tis","Tish","Tiz","Tys","Tysh","Ung","Ungr"];
var nm5 = ["o","a","i","ia","io","ou","ua","ah","um","oh","ir","ar","or","un","eo","ea","in","im","an","am","","","","",""];
var nm6 = ["leil","nar","nyl","myl","lyl","dyl","ris","ras","res","ros","win","wyn","waen","whin","whyn","whaen","tin","tyn","lyn","lin","lynn","linn","wynn","va","van","rin","ryn","ryl","nas","nan","ria","nia","ri","rip","nip","dora","leen","len","ma","la","mora","mura","mera","nura","nera","nora","glia","glian","giel","thiel","diel","thel","nis","niss","nys","nyss"];
var nm7 = ["Barley","Battle","Black","Bone","Boulder","Bright","Bronze","Cask","Cliff","Cold","Crag","Dark","Deep","Dirge","Doom","Fire","Flint","Frost","Fuse","Gold","Hammer","High","Iron","Long","Marble","Molten","Mountain","Pale","Red","Slate","Snow","Steel","Stern","Stone","Storm","Stout","Thunder"];
var nm8 = ["arm","axe","beard","belly","blade","braid","branch","brand","breaker","brew","brow","crag","dust","fall","fist","flayer","forge","fury","gem","grip","hammer","hand","helm","mane","mantle","ore","pike","river","roar","rock","shaper","shield","shout","steel","stone","toe"];

function nameGen(type){
	var tp = type;
	var br = "";
	var element = document.createElement("div");
	element.setAttribute("id", "result");
	
	for(i = 0; i < 10; i++){
		rnd4 = Math.floor(Math.random() * nm7.length);
		rnd5 = Math.floor(Math.random() * nm8.length);
		if(tp === 1){
			rnd = Math.floor(Math.random() * nm4.length);
			rnd2 = Math.floor(Math.random() * nm5.length);
			rnd3 = Math.floor(Math.random() * nm6.length);
			names = nm4[rnd] + nm5[rnd2] + nm6[rnd3] + " " + nm7[rnd4] + nm8[rnd5];
		}else{
			rnd = Math.floor(Math.random() * nm1.length);
			rnd2 = Math.floor(Math.random() * nm2.length);
			rnd3 = Math.floor(Math.random() * nm3.length);
			names = nm1[rnd] + nm2[rnd2] + nm3[rnd3] + " " + nm7[rnd4] + nm8[rnd5];
		}
		br = document.createElement('br');	
		element.appendChild(document.createTextNode(names));
		element.appendChild(br);
	}
	if(document.getElementById("result")){
		document.getElementById("placeholder").removeChild(document.getElementById("result"));
	}		
	document.getElementById("placeholder").appendChild(element);
}